src/NServiceBus.Metrics/Reporters/NServiceBusMetricReport.cs

```diff
@@ -26,17 +27,32 @@ public NServiceBusMetricReport(IDispatchMessages dispatcher, string destination)
 
         public void RunReport(MetricsData metricsData, Func<HealthStatus> healthStatus, CancellationToken token)
         {
+            RunReportAsync(metricsData)
+                .IgnoreContinuation();
+        }
+
+        async Task RunReportAsync(MetricsData metricsData)
+        {
             var serialized = JsonBuilderV2.BuildJson(metricsData);
             var body = Encoding.UTF8.GetBytes(serialized);
 
-            var headers = new Dictionary<string, string>();
+            var headers = new Dictionary<string, string>
+            {
+                [Headers.EnclosedMessageTypes] = "Metrics.Json.JsonMetricsContext, Metrics"
```

Comment:
Nope, because via reflection you would get `NServiceBus.Metrics` as assembly name.